### PR TITLE
setup datadog json logger and tracing lib

### PR DESCRIPTION
**Pull request by btkostner**
This

- adds JSON logging in production for better structured logs
- removes duplicate logger metadata
- adds datadog tracing so we can track function times and errors

Tested and working in staging. This will need updates to config for the `ENVIRONMENT` value to be set

### CODE REVIEW

File: config/releases.exs [hunk 1]
```diff
@@ -18,7 +18,9 @@ config :bullhorn,
 
 config :appsignal, :config,
   push_api_key: bullhorn_config["APPSIGNAL_PUSH_KEY"],
-  env: bullhorn_config["APPSIGNAL_ENV"]
+  env: bullhorn_config["ENVIRONMENT"]
+
+config :bullhorn, Bullhorn.Tracer, env: bullhorn_config["ENVIRONMENT"]
```
Comment (dbernheisel) [inline config/releases.exs:23]: will changing to `ENVIRONMENT` make staging and production report under the same env?

File: lib/bullhorn/application.ex [hunk 2]
```diff
@@ -9,6 +9,7 @@ defmodule Bullhorn.Application do
 
   def start(_type, _args) do
     children = [
+      {SpandexDatadog.ApiServer, [http: HTTPoison, host: "127.0.0.1"]},
```
Comment (dbernheisel) [inline lib/bullhorn/application.ex]: WDYT of moving the host, or maybe the options here entirely, to config.exs and releases.exs?

File: lib/bullhorn/orders.ex [hunk 3]
```diff
@@ -8,8 +8,6 @@ defmodule Bullhorn.Orders do
   alias Bullhorn.Mailer
 
   def tribble_failed(%TribbleFailed{order: order, type: failure_type}) do
-    Logger.metadata(order_id: order.id)
-
```
Comment (dbernheisel) [inline lib/bullhorn/orders.ex:12]: Any harm in leaving this in? or is it duplicated at this point? Feels like more logging is better logging

File: config/releases.exs (hunk 1 above)
Comment (dbernheisel) [reply config/releases.exs:23]: looked it up in our infra: `ENVIRONMENT` will be separate with `staging` and `production`. 👍

File: lib/bullhorn/orders.ex (hunk 3 above)
Comment (btkostner) [reply lib/bullhorn/orders.ex:12]: It's a duplicate at this point. All of the `order_id` and `user_id` will be set when we first read the message. That way we can make sure any error that happens between then has the needed metadata.

File: lib/bullhorn/application.ex (hunk 2 above)
Comment (btkostner) [reply lib/bullhorn/application.ex]: This configuration is static and shouldn't change so I moved it to `config/config.exs` :+1:

**Comment by btkostner** [comment]
Ok. I'm actually done with edits now. Everything works. I'm going to start including this in the other projects now.

**Review by doomspork — APPROVED** [review]

**Review by dbernheisel — APPROVED** [review]

**Pull request merged**